arbiter-core/src/manager.rs

```diff
@@ -8,40 +14,121 @@ use log::{info, warn};
+        label: String,
+    },
 
+    /// Indicates that no [`Environment`] exists with the provided label.
     #[error("environment labeled {label} does not exist!")]
-    EnvironmentDoesNotExist { label: String },
+    EnvironmentDoesNotExist {
+        /// The `label` that could not be found in the [`Manager`]'s collection.
+        label: String,
+    },
 
+    /// Indicates that the [`Environment`] with the given label is currently
+    /// running.
     #[error("environment labeled {label} is already running!")]
-    EnvironmentAlreadyRunning { label: String },
+    EnvironmentAlreadyRunning {
+        /// The `label` of the [`Environment`] that is already running.
+        label: String,
+    },
 
+    /// Indicates that the [`Environment`] with the given label is not running.
     #[error("environment labeled {label} is already stopped!")]
-    EnvironmentNotRunning { label: String },
+    EnvironmentNotRunning {
+        /// The `label` of the [`Environment`] that is not currently running.
+        label: String,
+    },
 
+    /// Indicates that the [`Environment`] with the given label has been stopped
+    /// and cannot be restarted or paused.
     #[error("environment labeled {label} has been stopped and cannot be restarted or paused!")]
-    EnvironmentStopped { label: String },
+    EnvironmentStopped {
+        /// The `label` of the [`Environment`] that is already stopped.
+        label: String,
+    },
 
+    /// Indicates that the [`Environment`] with the given label is currently
+    /// paused.
     #[error("environment labeled {label} is already paused!")]
-    EnvironmentAlreadyPaused { label: String },
+    EnvironmentAlreadyPaused {
+        /// The `label` of the [`Environment`] that is already paused.
+        label: String,
+    },
+
+    /// Indicates that the [`Environment`]'s thread handle could not be found.
+    #[error("no handle available to join the environment")]
+    NoHandleAvailable,
+
+    /// Indicates that the [`Environment`]'s thread has panicked.
+    #[error("joining on the environment thread resulted in a panic")]
+    ThreadPanic,
 }
 
 impl Manager {
+    /// Creates a new [`Manager`] with an empty set of environments.
     pub fn new() -> Self {
         Self {
             environments: HashMap::new(),
         }
     }
 
+    /// Adds a new environment to the manager with the specified label.
+    ///
+    /// This method creates a new environment with given parameters and
+    /// associates it with a label.
+    ///
+    /// # Parameters
+    ///
+    /// - `environment_label`: The label (identifier) to be used for the
+    ///   environment.
+    /// - `block_rate`: The mean of the rate at which the environment will
+    ///   process blocks (e.g., the rate parameter in the Poisson distribution
+    ///   used in the [`SeededPoisson`] field of an [`Environment`]).
+    /// - `seed`: A seed value for the environment, often used for randomness.
```

Comment:
Maybe we should just say 
    ```/// - `seed`: A seed value for the environment, used for randomness.```
 rather than
 ```   /// - `seed`: A seed value for the environment, often used for randomness.```
 to be precise

Reply:
How about:

`/// - `seed`: A value chosen to generate randomly chosen block sizes for the environment.`